### PR TITLE
engine: bugfix for user switching

### PR DESCRIPTION
**Pull request by maxtaco**
intro'ed in bdfde6aed1f859b91b6a8bbdd8e9626ab6ebd723

### CODE REVIEW

**Comment by maxtaco** [comment]
PTAL: @oconnor663

File: go/engine/login_current_device.go [hunk 1]
```diff
@@ -44,6 +44,10 @@ func (e *LoginCurrentDevice) SubConsumers() []libkb.UIConsumer {
 
 // Run starts the engine.
 func (e *LoginCurrentDevice) Run(ctx *Context) error {
+	e.G().Log.Debug("+- LoginCurrentDevice.Run")
+	defer func() {
+		e.G().Log.Debug("- LoginCurrentDevice.Run")
```
Comment (maxtaco) [inline go/engine/login_current_device.go]: debugging, sweet sweet debugging!

File: go/engine/login_current_device.go [hunk 2]
```diff
@@ -67,21 +79,19 @@ func (e *LoginCurrentDevice) Run(ctx *Context) error {
...
-		PublicKeyOptional: true,
-		ForceReload:       true,
-	})
+	me, err := libkb.LoadUser(loadUserArg)
 	if err != nil {
 		e.G().Log.Debug("error loading user profile: %#v", err)
 		return err
 	}
-	if !me.HasDeviceInCurrentInstall() {
+	if !me.HasDeviceInCurrentInstall(did) {
```
Comment (maxtaco) [inline go/engine/login_current_device.go]: can't pull the global device here, have to pull the one you want to talk about

File: go/engine/login_current_device.go [hunk 3]
```diff
@@ -67,21 +79,19 @@ func (e *LoginCurrentDevice) Run(ctx *Context) error {
 		e.G().Log.Debug("user config is nil")
 		return errNoConfig
 	}
-	if config.GetDeviceID().IsNil() {
+	did := config.GetDeviceID()
```
Comment (oconnor663) [inline go/engine/login_current_device.go]: can we call this variable "deviceID" or something. It took me a second to figure out what it meant :)

**Comment by oconnor663** [comment]
Looks good!

**Pull request merged**